homeassistant/components/recorder/migration.py

```diff
@@ -1186,6 +1212,69 @@ def _migrate_statistics_columns_to_timestamp(
                     )
 
 
+def _context_id_to_bytes(context_id: str | None) -> bytes | None:
+    """Convert a context_id to bytes."""
+    if context_id is None:
+        return None
+    if len(context_id) == 32:
+        return UUID(context_id).bytes
+    if len(context_id) == 26:
+        return ulid_to_bytes(context_id)
```

Comment:
This needs a test to make sure we can handle empty context_ids, the old format 36 byte uuids, and the 26 byte ulids